app/javascript/mastodon/components/media_gallery.js

```diff
@@ -85,14 +85,24 @@ class Item extends React.PureComponent {
     let thumbnail = '';
 
     if (attachment.get('type') === 'image') {
+      const previewUrl = attachment.get('preview_url');
+      const previewWidth = attachment.getIn(['meta', 'small', 'width']);
+
+      const originalUrl = attachment.get('remote_url') || attachment.get('url');
```

Comment:
You probably *don't* want to load remote_url just like this - it could be HTTP (while from HTTPS), and it would reveal the user's IP address to a remote server. So we've kinda decided against hotlinking.